code/modules/projectiles/firing.dm

```diff
@@ -39,67 +39,24 @@
 	var/turf/curloc = get_turf(user)
 	if (!istype(targloc) || !istype(curloc) || !BB)
 		return 0
-	BB.ammo_casing = src
 
 	var/firing_dir
 	if(BB.firer)
 		firing_dir = BB.firer.dir
 	if(!BB.suppressed && firing_effect_type)
 		PoolOrNew(firing_effect_type, list(get_turf(src), firing_dir))
 
+	var/direct_target
 	if(targloc == curloc)
-		if(target) //if the target is right on our location we go straight to bullet_act()
-			target.bullet_act(BB, BB.def_zone)
-		qdel(BB)
-		BB = null
-		return 1

```

Comment:
moved this piece of code inside projectile/proc/fire() (via the use of a second argument called "direct_target") for consistency.